controllers/grid/settings/roles/UserGroupGridHandler.inc.php

```diff
@@ -251,7 +251,11 @@ function initFeatures($request, $args) {
 	 * @param $request PKPRequest
 	 */
 	function addUserGroup($args, $request) {
-		return $this->editUserGroup($args, $request);
+		$userGroupForm = $this->_getUserGroupForm($request);
+
+		$userGroupForm->initData();
+
+		return new JSONMessage(true, $userGroupForm->fetch($request));
```

Comment:
Was this part necessary? I can't see how this changes anything from `$this->editUserGroup($args, $request)`.

Reply:
yes, it is actually the same as editUserGroup, just that the editUserGroup is defined as `userGroupRequiredOps` in the function `authorize`. I thought I would leave it there because it really requires userGourpId, and this addUserGourp function does not need it...
the other solution would be to remove the editUserGroup from `userGroupRequiredOps` i.e. maybe to totally remove that  check for userGroupId from the `authorize` function?

Reply:
I think that the authorization check is only applied when routing a request to a specific method. Once `addUserGroup()` is called, the authorization check is passed and you can call `$this->editUserGroup()` without invoking the additional authorization checks. Double-check that it works, but I think it should.

Reply:
I checked it, and it seems to be different -- when editUserGroup is called the authorize (and that ckeck) is called again...

Reply:
ah sorry, I was on OJS that did not have the change :-P

Reply:
ok, then I will only remove the updateGroupId form authorize...

Reply:
Once that's done, add a submodule commit for OJS to run the tests just in case. Otherwise, looks ready to go. :+1: